client/client.go

```diff
@@ -136,3 +141,39 @@ func (c *vpnClient) Resume() {
 		c.boxService.pauseManager.NetworkWake()
 	}
 }
+
+// ServerConnectConfig represents configuration for connecting to a custom server.
+type ServerConnectConfig []byte
+
+// SelectCustomServer replace box service instance by a instance using the
+// given config. If the Box service is already running, you'll need to
+// stop and start the VPN again so it can use the new instance.
+func (c *vpnClient) SelectCustomServer(cfg ServerConnectConfig) error {
+	options, err := json.UnmarshalExtended[option.Options]([]byte(cfg))
+	if err != nil {
+		return fmt.Errorf("unmarshal config: %w", err)
+	}
+
+	customInstance, err := box.New(box.Options{
+		Options: options,
```

Comment:
I think this has to be merged with our existing options

Reply:
I've initially tried to do that but some things appeared that made me create a new instance. Currently sing-box don't support adding Routing rules to a created instance, we're able to add Inbounds and Outbounds when a instance is running, but without updating the routing rules the traffic wouldn't be sent to the selected server.

Reply:
No I'm not saying we add it to the existing instance. I'm saying we need to include the previous run options with these new options before restarting.